common/src/configs/mod.rs

```diff
@@ -1,10 +1,89 @@
+pub mod audio_config;
+pub mod display_config;
+pub mod model_config;
+pub mod physics_config;
+pub mod player_config;
+pub mod scene_config;
+pub mod texture_config;
+
+use once_cell::sync::Lazy as OnceCellLazy;
 use serde::Serialize;
 use std::fs::File;
 use std::io::{Read, Write};
 use std::path::Path;
+use std::sync::{Arc, RwLock};
 
-pub mod model_config;
-pub mod scene_config;
+use crate::configs::audio_config::ConfigAudioAssets;
+use crate::configs::display_config::ConfigDisplay;
+use crate::configs::model_config::ConfigModels;
+use crate::configs::player_config::ConfigPlayer;
+use crate::configs::scene_config::ConfigSceneGraph;
+use crate::configs::texture_config::ConfigTexture;
+
+const MODELS_CONFIG_PATH: &str = "models.json";
+const SCENE_CONFIG_PATH: &str = "scene.json";
+const AUDIO_CONFIG_PATH: &str = "audio.json";
+const PLAYER_CONFIG_PATH: &str = "player.json";
+const DISPLAY_CONFIG_PATH: &str = "display.json";
+const TEXTURE_CONFIG_PATH: &str = "tex.json";
+
+pub static CONFIG_INSTANCE: OnceCellLazy<RwLock<Option<Arc<Config>>>> =
+    OnceCellLazy::new(|| RwLock::new(None));
+
+pub struct Config {
+    pub models: ConfigModels,
+    pub scene: ConfigSceneGraph,
+    pub audio: ConfigAudioAssets,
+    pub player: ConfigPlayer,
+    pub display: ConfigDisplay,
+    pub texture: ConfigTexture,
+}
+
+impl Config {
+    pub fn new(
+        models: ConfigModels,
+        scene: ConfigSceneGraph,
+        audio: ConfigAudioAssets,
+        player: ConfigPlayer,
+        display: ConfigDisplay,
+        texture: ConfigTexture,
+    ) -> Self {
+        Config {
+            models,
+            scene,
+            audio,
+            player,
+            display,
+            texture,
+        }
+    }
+}
+
+pub struct ConfigurationManager;
```

Comment:
not important but if the struct is empty, we can just use a `mod` to bundle a bunch of static functions